### DISCUSSION TITLE
drop edge

### DISCUSSION THREAD

**Discussion by wubo2180** [pyg-team/pytorch_geometric#4210] [Q&A] [answered]
Hi, I would want to know is there any method to realize the dropping edge as well as removing the dropped edges' attribute?

**Comment by rusty1s** [comment, marked as answer]
Please take a look at `utils.dropout_adj`. Might exactly be what you need :)

**Reply by wubo2180** [reply]
thanks! I would want to know is there any method to realize the dropping the nodes?

**Reply by rusty1s** [reply]
You can randomly generate a node mask and operate on the resulting subgraph:
```python
node_mask = torch.rand(data.num_nodes) > 0.5
data = data.subgraph(node_mask)
```
